### PR TITLE
Maxim dev

### PR DESCRIPTION
**Pull request by imefistoff**

### CODE REVIEW

File: app/src/main/java/com/example/sportsclubmanagement/screens/main/fragments/home/HomeFragment.java [hunk 1]
```diff
@@ -124,6 +118,44 @@ public void onFailure(Call<UserDetails> call, Throwable t) {
...
+            public void onFailure(Call<List<EventsAvailable>> call, Throwable t) {
+                Toast.makeText(getContext(), t.toString(), Toast.LENGTH_LONG).show();
+                call.cancel();
+            }
+        });
+    }
+
+    private void populateRecyclerEvents() {
+        //first evetns
+        firstEventAdapter = new EventAdapter(events, this.getContext(), this);
```
Comment (IonutDoroftei) [inline app/src/main/java/com/example/sportsclubmanagement/screens/main/fragments/home/HomeFragment.java:149]: You can add a function that create instances for each adapter

File: app/src/main/java/com/example/sportsclubmanagement/screens/main/fragments/home/HomeFragment.java [hunk 2]
```diff
@@ -124,6 +118,44 @@ public void onFailure(Call<UserDetails> call, Throwable t) {
...
+                if (response.isSuccessful()) {
+                    for (EventsAvailable event : response.body()) {
+                        events.add(new EventAdapterModel(event.getId(), event.getTitle(), event.getLocatia(), event.getDate(), true, false));
+                        future.add(new EventAdapterModel(event.getId(), event.getTitle(), event.getLocatia(), event.getDate(), true, true));
+                    }
+                    populateRecyclerEvents();
+                    Log.d("Hom", "List of events was accepted");
+                } else {
+                    Log.d("Home", "Error");
+                    Toast.makeText(getContext(), "Fail gave all events", Toast.LENGTH_SHORT).show();
```
Comment (IonutDoroftei) [inline app/src/main/java/com/example/sportsclubmanagement/screens/main/fragments/home/HomeFragment.java:135]: You can add all toast messages on constants

File: app/src/main/java/com/example/sportsclubmanagement/screens/main/fragments/home/HomeFragment.java [hunk 3]
```diff
@@ -52,6 +51,8 @@
     private APIInterface apiInterface;
     private SharedPreferences pref;
     private SharedPreferences.Editor editor;
+    private List<EventAdapterModel> events;
```
Comment (IonutDoroftei) [inline app/src/main/java/com/example/sportsclubmanagement/screens/main/fragments/home/HomeFragment.java:54]: Check if you really need to make this global variable

File: app/src/main/java/com/example/sportsclubmanagement/screens/eventdetails/EventActivity.java [hunk 4]
```diff
@@ -13,23 +17,80 @@
...
+
+    private void fillEventDetailsOnPage() {
+        title_event.setText(eventDetails.getTitle());
+        locationEvent.setText(eventDetails.getLocation());
+        SimpleDateFormat formatterTime = new SimpleDateFormat("HH:mm", Locale.ENGLISH);
+        timeEvent.setText(formatterTime.format(eventDetails.getTimeEvent()));
+        SimpleDateFormat formatter= new SimpleDateFormat("dd.MM.yyyy", Locale.ENGLISH);
+        dateEvent.setText(formatter.format(eventDetails.getDataEvent()));
+        //cand am imaginea de pe back
+        //Glide.with(this.getApplicationContext()).load().centerCrop().into(imgEvent);
```
Comment (IonutDoroftei) [inline app/src/main/java/com/example/sportsclubmanagement/screens/eventdetails/EventActivity.java:87]: remove lines of code that are commented

File: app/src/main/java/com/example/sportsclubmanagement/rest/APIClient.java [hunk 5]
```diff
@@ -13,11 +16,14 @@ public static Retrofit getClient() {
         HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
         interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
         OkHttpClient client = new OkHttpClient.Builder().addInterceptor(interceptor).build();
+        Gson gson = new GsonBuilder()
```
Comment (IonutDoroftei) [inline app/src/main/java/com/example/sportsclubmanagement/rest/APIClient.java:19]: I'm not sure you really need this converter

File: app/src/main/java/com/example/sportsclubmanagement/screens/main/fragments/home/HomeFragment.java (hunk 1 above)
Comment (imefistoff) [reply app/src/main/java/com/example/sportsclubmanagement/screens/main/fragments/home/HomeFragment.java]: done

File: app/src/main/java/com/example/sportsclubmanagement/screens/main/fragments/home/HomeFragment.java (hunk 2 above)
Comment (imefistoff) [reply app/src/main/java/com/example/sportsclubmanagement/screens/main/fragments/home/HomeFragment.java]: done

File: app/src/main/java/com/example/sportsclubmanagement/rest/APIClient.java (hunk 5 above)
Comment (imefistoff) [reply app/src/main/java/com/example/sportsclubmanagement/rest/APIClient.java]: done

**Pull request merged**